src/hardware/mixer.cpp

```diff
@@ -790,17 +846,17 @@ void MixerChannel::AddStretched(uint16_t len, int16_t *data)
 		const auto new_pos = index >> FREQ_SHIFT;
 		if (pos != new_pos) {
 			pos = new_pos;
-			//Forward the previous sample
-			prev_sample[0] = data[0];
+			// Forward the previous sample
+			prev_frame.left = data[0];
 			data++;
```

Comment:
This is unrelated to the PR, but I suspect there's bug here.
If we pass a tiny array, say, 

`int16_t data[1] = {0};` into 

into 
 `AddStretched(1, data)`

Then this loop-logic here marches `data++` forward without any concern for the `len`, and instead the loop amount is driven by `auto outlen = needed - done;`, none of which are affected by the incoming data `len`gth. 

(wanted to just get this typed down while I was still looking at this chunk).

Reply:
Yeah I suggested in one of my earlier comments to get rid of `AddStretched` completely; it's only used a single time in `sblaster.cpp` and I suspect it's not really needed.